frontend/src/metabase/visualizations/visualizations/SmartScalar/SettingsComponents/AnotherColumnForm.tsx

```diff
@@ -96,3 +98,29 @@ export function AnotherColumnForm({
     </Box>
   );
 }
+
+function getInitialColumn(
+  value: SmartScalarComparisonAnotherColumn | undefined,
+  columns: DatasetColumn[],
+) {
+  if (value?.column) {
+    return value.column;
+  }
+  if (columns.length === 1) {
+    return columns[0].name;
+  }
+  return "";
+}
+
+function getInitialLabel(
```

Comment:
Nit: it would be nice to combine these two functions into one to keep the logic together instead of decoupled, for example:

```typescript
  const { initialColumn, initialLabel } = getInitialColumnAndLabel(
    selectedValue,
    columns,
  );

  const [label, setLabel] = useState(initialColumn);
  const [column, setColumn] = useState(initialLabel);
  
  ...
  
  function getInitialColumnAndLabel(
  value: SmartScalarComparisonAnotherColumn | undefined,
  columns: DatasetColumn[],
) {
  if (value?.column) {
    return { initialColumn: value.column, initialLabel: value.label };
  }
  if (columns.length === 1) {
    return {
      initialColumn: columns[0].name,
      initialLabel: columns[0].display_name,
    };
  }
  return { initialColumn: "", initialLabel: "" };
}

  
  ```